dask/dataframe/core.py

```diff
@@ -4328,21 +4328,19 @@ def apply(self, func, convert_dtype=True, meta=no_default, args=(), **kwds):
         --------
         dask.Series.map_partitions
         """
+        if convert_dtype is not no_default:
+            kwds["convert_dtype"] = convert_dtype
+
+        # let pandas trigger any warnings, such as convert_dtype warning
+        self._meta_nonempty.apply(func, args=args, **kwds)
+
         if meta is no_default:
             meta = _emulate(
-                M.apply,
-                self._meta_nonempty,
-                func,
-                convert_dtype=convert_dtype,
-                args=args,
-                udf=True,
-                **kwds,
+                M.apply, self._meta_nonempty, func, args=args, udf=True, **kwds
```

Comment:
I'm surprised this doesn't need a ` with check_convert_dtype_deprecation()`. Maybe we're emitting multiple, duplicate warnings?

Reply:
Good point, I'll see how many we're emitting.

Reply:
Yes, it was two, only in case there's no meta, which is why I missed it. Fixed now.